include/knowhere/index/handle.h

```diff
@@ -0,0 +1,60 @@
+// Copyright (C) 2019-2023 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under the License.
+#ifndef KNOWHERE_HANDLE_H
+#define KNOWHERE_HANDLE_H
+#include <atomic>
+#include <chrono>
+#include <memory>
+
+#include "knowhere/expected.h"
+
+namespace folly {
+template <typename T>
+class Future;
+
+}
+
+namespace knowhere {
+class Handle {
+ public:
+#ifdef KNOWHERE_WITH_CARDINAL
+    explicit Handle(const std::chrono::seconds& timeout);
+
+    void
+    Stop();
+
+    bool
+    Flag() const;
+
+    bool
+    IsTimeout() const;
+#else
+    Handle();
+#endif
+
+    Status
+    Get();
+
+    void
+    Set(folly::Future<Status>&& future);
+
+    ~Handle();
+
+ private:
+#ifdef KNOWHERE_WITH_CARDINAL
+    std::chrono::steady_clock::time_point start;
+    std::chrono::seconds timeout;
+    mutable std::atomic_bool flag = false;
+#endif
+    std::unique_ptr<folly::Future<Status>> future = nullptr;
```

Comment:
same here. The name `future` is way too generic. Is it a handler, a callback, just a future? Also, please add some comments on what this is.